### PR TITLE
#180: add exit and help cmds

### PR DESCRIPTION
**Pull request by ab22**
## Description

Changes for issue #180 

The `help` and `exit` commands work correctly when typing them, however, when typing the new commands the suggestions list won't show up and this is because of the way their [previous commands were added](https://github.com/ab22/genji/blob/master/cmd/genji/shell/command.go#L21-L22).

Also, when typing `.help` or `.exit` it now displays `.help or help` or `.exit or exit` on the suggestions. I am wondering how would you like these commands to be added to the suggestion list. Maybe an `alias []string` under the `commands` structure?

### CODE REVIEW

**Comment by ab22** [comment]
Would you guys like me to propose code changes to correctly support the new command suggestions correctly? @asdine

**Comment by asdine** [comment]
@ab22 Thanks for taking care of this!

> Maybe an alias []string under the commands structure?

I think this is a good idea

File: cmd/genji/shell/command.go [hunk 1]
```diff
@@ -116,26 +134,36 @@ func runHelpCmd() error {
...
+		for _, alias := range c.Aliases {
+			if shouldDisplaySuggestion(alias, in) {
+				suggestions = append(suggestions, alias)
+			}
+		}
 	}
 
 	if len(suggestions) == 0 {
-		return fmt.Errorf("Unknown command %q. Enter \".help\" for help.", in)
+		return fmt.Errorf("unknown command %q. Enter \".help\" for help", in)
```
Comment (ab22) [inline cmd/genji/shell/command.go:166]: This fixes a go-lint issue: `error strings should not be capitalized or end with punctuation or a newline`

**Comment by ab22** [comment]
I went ahead and added a small command builder. This way we add a little more flexibility for the commands in the future, for example if we wanted to add individual options instead of just a string, then this would easily perform validations when building it.

File: cmd/genji/shell/command.go [hunk 2]
```diff
@@ -13,16 +13,34 @@ import (
...
-	{".exit or exit", ``, "Exit this program."},
-	{".help or help", ``, "List all commands."},
-	{".tables", ``, "List names of tables."},
-	{".indexes", `[table_name]`, "Display all indexes or the indexes of the given table name."},
-	{".dump", `[table_name]`, "Dump database content or table content as SQL statements"},
+	Aliases     []string
+}
+
+var commands = []*command{
+	newCommandBuilder(".exit").
```
Comment (asdine) [inline cmd/genji/shell/command.go:25]: This feels very Java-esque, I feel the same can be achieved by filling the struct directly:

```go
var commands = []*command{
  {
    name: ".exit",
    aliases: []string{"exit"},
    description: "Exit this program.",
  },
  {
    name: ".help",
    aliases: []string{"help"},
    description: "List all commands.",
  },
}

type command struct {
	Name        string
	Options     string
	DisplayName string
	Description string
}

func (c *command) String() string {
  ...
}
```

File: cmd/genji/shell/command.go [hunk 3]
```diff
@@ -16,13 +16,45 @@ import (
...
+	{
+		Name:        ".help",
+		Options:     "",
+		DisplayName: ".help or help",
+		Description: "List all commands.",
+		Aliases:     []string{"help"},
+	},
+	{
+		Name:        ".tables",
+		Options:     "",
```
Comment (asdine) [inline cmd/genji/shell/command.go:39]: nit: You don't have to do this in Go, non initialized fields are equal to the [zero value of their type](https://www.geeksforgeeks.org/zero-value-in-golang/)

```suggestion
```

File: cmd/genji/shell/command.go [hunk 4]
```diff
@@ -16,13 +16,45 @@ import (
...
+		DisplayName: ".tables",
+		Description: "List names of tables.",
+		Aliases:     []string{},
```
Comment (asdine) [inline cmd/genji/shell/command.go:42]: Same here, you can remove this line

**Review by asdine — CHANGES_REQUESTED** [review]
Thanks @ab22, this looks good, I just have a few nitpicking comments

**Comment by ab22** [comment]
Hey @asdine, sure, no problem. I left those out because of how they were previously declared. Just forced pushed a commit to remove those lines. :D

**Review by asdine — APPROVED** [review]
Thank you @ab22 !

**Pull request merged**